ddl/ddl_api.go

```diff
@@ -1192,6 +1191,19 @@ func checkPartitionByRange(ctx sessionctx.Context, tbInfo *model.TableInfo, pi *
 	return nil
 }
 
+func checkPartitionByRangeColumn(ctx sessionctx.Context, tbInfo *model.TableInfo, pi *model.PartitionInfo, s *ast.CreateTableStmt) error {
+	if err := checkPartitionNameUnique(tbInfo, pi); err != nil {
+		return err
+	}
+
+	// TODO: Check CreatePartitionValue.
```

Comment:
Range columns partition key supports multiple data types with integer、datetime、string.

Reply:
Yes, I know, and I'll do it later.
https://github.com/pingcap/tidb/issues/9083

Reply:
Do we need to check `validRangePartitionType`?

Reply:
Range partition type is checked.
This is the new added range columns partition.